verified_boot.md

```diff
@@ -0,0 +1,240 @@
+  taking control away from users?
+  - Security: A guarantee that only firmware from a trusted vendor is run
+  - Control: Ability to self-sign, to inspect and replace firmware components
+- There are other verified boot schemes
+
+---
+
+# Intel Boot Guard
+
+- The most widely known option
+- Verifies and measures the initial bootblock
+- Ensures FW authenticity using keys fused to the chipset
+- Supports firmware verification and measurement
+- Different profiles with different features enabled
+  - Verified, Measured and Enforced knobs
+  - Verified boot is always enabled in all profiles
+- Ties a platform to a specific firmware vendor, forever
+
+???
+
+- Intel specific, AMD has a different but comparable secure boot scheme
+  Verifies the initial bootblock, rest is up to BIOS (Vboot)
+- fused: programmed into one-time programmable e-fuses inside the chipset
+- Measurements are made in locality 3, so they can't be faked by malicious BIOS
+- Profile 3: Verified and Measured with infinite time for remediation
+- Profile 3 acts as root of trust for measurement, but not for verification
+- On verification failure, the PCR0 measurements are always the same. This was
+  determined experimetally.
+- so users can choose to use Dasharo firmware and get measured IBB, or flash
+  their own and lose measured IBB.
+  - might make sense for some small percentage for users
+  - does not help us a lot
+
+---
+
+# Boot Guard cons
+
+- Requires more blobs in firmware
+  - Boot Guard ACM
+- Fusing removes some owner control
+- Profile 3 does not ensure initial boot block authenticity
+  - Only helps us establish a root of trust for measurement
+- Other profiles take away owner control completely
+
+???
+
+- ACM - Authenticated Code Module. Hyper-privileged, Intel signed blob that
+  performs the actual BIOS verification and measurement, then hands off to the
+  BIOS reset vector.
+
+---
+
+# CBFS Verification
+
+- A relatively new coreboot feature
+- Cryptographically verifies components of the coreboot image
+- According to documentation:
+  > This only makes sense if you use some
+  > out-of-band mechanism to guarantee the integrity of the bootblock
+  > itself, such as Intel Boot Guard or flash write-protection.
+- Depends on some other mechanism for signing (e.g. Boot Guard)
+- Can effectively replace the portions of Vboot that Dasharo uses
+
+???
+
+- CBFS - coreboot's internal file system
+
+---
+
+# Firmware updates
+
+- Chrome Vboot: A+B+RO
+  - A and B slots for updates, RO slot contains IBB and verification code
+  - Anti-rollback using TPM
+  - Vboot aware OS service manages slot updates
+- Dasharo:
+  - Initial idea was to leave RO untouched and only update A/B slots
+  - In practice, most updates introduce breaking changes that require updating
+    the bootblock
+  - Dasharo Tools Suite handles updates, does the entire update in one operation
+    - So we don't use the A/B feature
+  - Capsule Updates will help here
```

Comment:
`Will` is too strong word, this is something that we may end up skipping.

```suggestion
  - Capsule Updates may help here
```

Reply:
https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR197